rust/agama-lib/src/security/model.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Serialize, Deserialize, utoipa::ToSchema)]
+pub enum SSLFingerprintAlgorithm {
+    SHA1,
+    SHA256,
+}
+
+impl TryFrom<String> for SSLFingerprintAlgorithm {
```

Comment:
To save some code, you might want to use https://docs.rs/strum/latest/strum/derive.EnumString.html from the [strum crate](https://crates.io/crates/strum). We are already using it in a few places.